### PR TITLE
Adding docs for dynamic registry apis

### PR DESCRIPTION
**Pull request by joshua-kim**
Ticket: https://ava-labs.atlassian.net/browse/AV-1308

### CODE REVIEW

**Comment by yulin-dong** [comment]
@joshua-kim we cannot merge this until after the code has been released, right?

**Comment by joshua-kim** [comment]
> @joshua-kim we cannot merge this until after the code has been released, right?

Yep, otherwise it'll be confusing to developers.

**Comment by yulin-dong** [comment]
LGTM

File: docs/build/avalanchego-apis/admin.md
Comment (ceyonur) [inline docs/build/avalanchego-apis/admin.md:208]: Can we put more information like where these plugin binaries should be put, in which form etc? We already have this information in https://docs.avax.network/build/tutorials/platform/subnets/create-a-virtual-machine-vm#executable-binary we can try to refer there but probably best to have a separate section (titled with AvalancheGo plugins etc) for the part:
> AvalancheGo searches for and registers plugins under [buildDir]/plugins/. You need to put built VM binary under this path. The [buildDir] defaults to the path of executed AvalancheGo binary. See [here](https://docs.avax.network/build/references/avalanchego-config-flags#build-directory) for more information. Executable names must be either a full VM ID (encoded in CB58), or must be a VM alias as explained below.

alternatively we can move this part under the https://docs.avax.network/build/references/avalanchego-config-flags/#build-directory and refer this section in other two pages (in VM page and in Admin API page)

Comment (yulin-dong) [reply docs/build/avalanchego-apis/admin.md:208]: @joshua-kim can you look over @ceyonur 's comment above and resolve it?

Comment (joshua-kim) [reply docs/build/avalanchego-apis/admin.md]: Added a sub-section under the VM installation tutorial

File: docs/build/avalanchego-apis/admin.md
Comment (ceyonur) [inline docs/build/avalanchego-apis/admin.md:237]: can we add another vm here without any aliases? So people won't freak out if they don't see anything in the array

File: docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md
Comment (ceyonur) [inline docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md:1472]: Can we add that it's also possible to install them by copying to plugin dir "before" node starts up?

File: docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md
Comment (ceyonur) [inline docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md:1467]:
```suggestion
```

File: docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md
Comment (ceyonur) [inline docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md:1488]:
```suggestion
```json
```

File: docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md
Comment (ceyonur) [inline docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md:1485]: do we need to specify jq here? if it's possible can we omit using additional commands?

File: docs/build/avalanchego-apis/info.md
Comment (ceyonur) [inline docs/build/avalanchego-apis/info.md:240]: I don't know the exact reason but we use `cpp` syntax for these signatures. We should definitely take another pass on these in-line code types...

File: docs/build/avalanchego-apis/admin.md
Comment (ceyonur) [inline docs/build/avalanchego-apis/admin.md:210]:
```suggestion
```cpp
```

File: docs/build/avalanchego-apis/info.md
Comment (ceyonur) [inline docs/build/avalanchego-apis/info.md:259]: IMO these should be `json` but it would probably be inconsistent with other places.

File: docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md
Comment (ceyonur) [inline docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md:1468]: noice 👍

File: docs/build/avalanchego-apis/admin.md
Comment (ceyonur) [reply docs/build/avalanchego-apis/admin.md]: 👍

File: docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md
Comment (joshua-kim) [reply docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md:1472]: Added

File: docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md
Comment (joshua-kim) [reply docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md:1467]: Bad vim typo. Removed.

File: docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md
Comment (joshua-kim) [reply docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md:1485]: No, just nice for formatting. Removed.

File: docs/build/avalanchego-apis/info.md
Comment (joshua-kim) [reply docs/build/avalanchego-apis/info.md:240]: Updated for consistency, but we should be using text imo. We can fix it in a separate pr.

File: docs/build/avalanchego-apis/admin.md
Comment (joshua-kim) [reply docs/build/avalanchego-apis/admin.md:210]: Updated as per previous comment

File: docs/build/avalanchego-apis/info.md
Comment (joshua-kim) [reply docs/build/avalanchego-apis/info.md:259]: Updated

File: docs/build/avalanchego-apis/info.md
Comment (ceyonur) [reply docs/build/avalanchego-apis/info.md:240]: yeah and I'm afraid it might be a HUUGE PR.

File: docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md
Comment (ceyonur) [inline docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md:1483]: Hm it's probably best to not direct them how to start their node. The page is swarmed with other information like wallet etc. Can we remove this line and just tell it's sufficient to copy binaries before startup? sorry for the nit 😶

Comment (joshua-kim) [reply docs/build/tutorials/platform/subnets/create-a-virtual-machine-vm.md]: Sure : )

File: docs/build/avalanchego-apis/info.md
Comment (yulin-dong) [inline docs/build/avalanchego-apis/info.md:282]:
```suggestion
            "jvYyfQTxGMJLuGWa55kdP2p2zSUYsQ5Raupu4TW34ZAUBAbtq": [
                "avm"
            ],
            "mgj786NP7uDwBCcq6YwThhaN8FLyybkCa4zBWTQbNgmK6k9A6": [
                "evm"
            ],
            "qd2U4HDWUvMrVUeTcCHp6xH3Qpnn1XbU5MDdnBoiifFqvgXwT": [
                "nftfx"
            ],
            "rWhpuQPF1kb72esV2momhMuTYGkEb1oL29pt2EBXWmSy4kxnT": [
                "platform"
            ],
            "rXJsCSEYXg2TehWxCEEGj6JU2PWKTkd6cBdNLjoe2SpsKD9cy": [
                "propertyfx"
            ],
            "spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ": [
                "secp256k1fx"
            ]
        }
```
Is this correct? Feel free to discard if not right.

**Review by ceyonur — APPROVED** [review]

File: docs/build/avalanchego-apis/info.md
Comment (joshua-kim) [reply docs/build/avalanchego-apis/info.md]: My IDE autoformats this weirdly. Updated.

**Pull request merged**
